net/instaweb/rewriter/resource.cc

```diff
@@ -269,17 +270,38 @@ void Resource::LinkFallbackValue(HTTPValue* value) {
 }
 
 StringPiece Resource::ExtractUncompressedContents() const {
-  ResponseHeaders headers;
-  if (!extracted_ && value_.ExtractHeaders(&headers, NULL)) {
-    if (headers.IsGzipped()) {
-      StringWriter inflate_writer(&extracted_contents_);
-      if (GzipInflater::Inflate(raw_contents(), GzipInflater::kGzip,
-                                &inflate_writer)) {
-        extracted_ = true;
-      }
+  bool use_extracted = EnsureExtractedIfNeeded();
+  return use_extracted ? extracted_contents_ : raw_contents();
+}
+
+const ResponseHeaders* Resource::UncompressedHeaders() const {
+  bool use_extracted = EnsureExtractedIfNeeded();
+  return use_extracted ? extracted_headers_.get() : response_headers();
+}
+
+bool Resource::EnsureExtractedIfNeeded() const {
+  if (extracted_state_ == kExtractNotComputed) {
```

Comment:
I wonder what happens when `response_headers_` gets changed after `extracted_state_` is computed.
If that is a valid case, perhaps this needs to reset `extracted_state_` to `kExtractNotComputed` when we observe response_headers_ is dirty / has been modified?

Reply:
Giflib is hosted on sourceforge, which seems to be very unstable in the last couple of months. Maybe we should set up a mirror for giflib on github.

Reply:
It's actually not that but rather the submodule registration stuff --- missing pngconf.h or such somehow.

Reply:
Ugh. Manually cp'ing third_party/libpng/src/scripts/pnglibconf.h.prebuilt to third_party/libpng/src/pnglibconf.h should work.

This happens because a build execution step isn't always executed timely. I run make with `-j8`  which seems to mask the problem. But with -j1 the problem does show on my system.


In https://github.com/apache/incubator-pagespeed-mod/pull/1739 I backed out two commits that attempted to fix that to avoid conflicting with Ashish his work:
- https://github.com/apache/incubator-pagespeed-mod/pull/1739/commits/2b4ae6032986654dc73ceee48e9f70de9267a9ad
- https://github.com/apache/incubator-pagespeed-mod/pull/1739/commits/f72e2111070dc07030e0390dfe07f7ae65d78826

Let me try with those again to see if they fix the problem.

Reply:
So giflib was fixed, CI is passing again on master, let me see if I can fix the copying of third_party/libpng/src/scripts/pnglibconf.h as well

Reply:
Ugh, I'm having a real problem fixing the problem with libpng failing on single-threaded builds. I have been at it all day, but no luck yet.

Apparently there is an ordering issue that gets masked when using -j in make. The problem is that the dependency declarations for libpng look good to me, yet somehow the action [1] responsible for copying a required header file isn't always running timely when it ought to.

[1] https://github.com/apache/incubator-pagespeed-mod/blob/master/third_party/libpng/libpng.gyp#L27